internal/backuprestore/backup.go

```diff
@@ -223,12 +223,20 @@ func (h *BRHandler) cleanupBackupLabels(ctx context.Context, backup *velerov1.Ba
 	if err != nil {
 		return fmt.Errorf("failed to get objs from apply-label annotations: %w", err)
 	}
+
 	// json patch escape info: https://jsonpatch.com/#json-pointer
 	escaped := strings.Replace(backupLabel, "/", "~1", 1)
 	payload := []byte(fmt.Sprintf(`[{"op":"remove","path":"/metadata/labels/%s"}]`, escaped))
+
 	for _, obj := range objs {
-		err := patchObj(ctx, h.DynamicClient, &obj, false, payload) //nolint:gosec
-		if err != nil {
+		patchedObj := obj
+
+		if hasBackupLabel, _ := searchBackupLabel(ctx, h.DynamicClient, &patchedObj); !hasBackupLabel {
+			h.Log.Info("backup label doesn't exist, no patching needed, ignoring", "name", obj.Name,
+				"namespace", obj.Namespace, "resource", obj.Resource, "group", obj.Group, "version", obj.Version)
+			continue
+		}
+		if err := patchObj(ctx, h.DynamicClient, &patchedObj, false, payload); err != nil {
```

Comment:
Or another option would be check the error type(k8s.IsInvalid-label doesn't exist and k8s.IsNotFound) of `patch` instead of making another `get` request

Reply:
yea, I was trying to avoid changing the `patchObj` function. Also, if we check for `k8serrors.IsNotFound(err) || k8serrors.IsInvalid(err)` seems like we can't log (silently ignored) when the label is not found

Reply:
gave it another spin in my head and decided to go with the checking errors approach you suggested Angie, it seems simpler indeed, thanks!